pandas/core/groupby/generic.py

```diff
@@ -1550,18 +1454,23 @@ def _insert_inaxis_grouper_inplace(self, result):
                 result.insert(0, name, lev)
 
     def _wrap_aggregated_output(self, output, names=None):
-        agg_axis = 0 if self.axis == 1 else 1
-        agg_labels = self._obj_with_exclusions._get_axis(agg_axis)
+        index = self.grouper.result_index
```

Comment:
I had to change the logic of this function around because it didn't really utilize the keys of an input dictionary as axis labels, nor did it do anything with `names` in spite of it being in the signature

I also found the following behavior of the DataFrame constructor interesting when trying to make this as generic as possible:

```python
>>> pd.DataFrame({"a": [1], "b": [2]})
   a  b
0  1  2
>>> pd.DataFrame({"a": [1], "b": [2]}, columns=["c", "d"])
Empty DataFrame
Columns: [c, d]
Index: []
```

The latter of that probably *should* raise but definitely a separate exercise (haven't checked for outstanding issues yet)

`index` had to be a part of the constructor to appropriately deal with empty groups still being part of the returned object

Reply:
this is as expected
passing columns is a reindex

Reply:
Makes sense. Had to change this around a bit anyway so I don't think original comment is as applicable - but definitely appreciate clarification